client/agent_endpoint.go

```diff
@@ -233,3 +233,124 @@ func (a *Agent) Host(args *structs.HostDataRequest, reply *structs.HostDataRespo
 	reply.HostData = data
 	return nil
 }
+func (a *Agent) monitorExport(conn io.ReadWriteCloser) {
+	defer conn.Close()
+
+	// Decode arguments
+	var args cstructs.MonitorExportRequest
+
+	decoder := codec.NewDecoder(conn, structs.MsgpackHandle)
+	encoder := codec.NewEncoder(conn, structs.MsgpackHandle)
+
+	if err := decoder.Decode(&args); err != nil {
+		handleStreamResultError(err, pointer.Of(int64(500)), encoder)
+		return
+	}
+
+	// Check acl
+	if aclObj, err := a.c.ResolveToken(args.AuthToken); err != nil {
+		handleStreamResultError(err, pointer.Of(int64(403)), encoder)
+		return
+	} else if !aclObj.AllowAgentRead() {
+		handleStreamResultError(structs.ErrPermissionDenied, pointer.Of(int64(403)), encoder)
+		return
+	}
```

Comment:
I sort of wonder if `agent:read` is too low because we're giving access to other service logs, but the [agent ACL](https://developer.hashicorp.com/nomad/docs/other-specifications/acl-policy#agent-rules) is actually pretty powerful already, so maybe that's ok? Might be worth reviewing with @dduzgun-security

Reply:
I've been thinking about this a lot and combined with the fact that nomad runs as root, I've been thinking it's probably best to attempt to ensure they're only reading from the nomad service.

As far as I can tell we don't really have any way of knowing if nomad is running as a systemd service, would it be too hamfisted to require that that the serviceName includes the word "nomad"? And either add that to the Scan or check that in the command itself.